plaso/parsers/ntfs.py

```diff
@@ -291,6 +548,16 @@ def ParseFileObject(self, parser_mediator, file_object):
       parser_mediator.ProduceExtractionWarning(
           'unable to open file with error: {0!s}'.format(exception))
 
+    # Collect path information in a first round of parsing
+    for entry_index in range(0, mft_metadata_file.number_of_file_entries):
+      try:
+        mft_entry = mft_metadata_file.get_file_entry(entry_index)
+        self._CollectMFTEntryPathInfo(mft_entry)
```

Comment:
doing this for every MFT entry can be memory intensive. Technically we only need to cache the parent path hints in `self.path_info`